game-core/src/test/java/games/strategy/triplea/UnitUtilsTest.java

```diff
@@ -125,5 +128,72 @@ void bombingDamageIsNotTransferredIfDamageAllowedIs0() {
           newUnits.get(0).getUnitDamage(),
           is(0));
     }
+
+    @Test
+    void unloadedUnitsAreTransferred() {
+      final Unit oldUnit = transport.create(1, player).get(0);
+      final List<Unit> newUnits = transport.create(1, player);
+
+      final List<Unit> unloadedInfantry = infantry.create(1, player);
+      oldUnit.setUnloaded(unloadedInfantry);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+      gameData.performChange(changes);
+
+      assertThat(newUnits.get(0).getUnloaded(), is(unloadedInfantry));
+    }
+
+    @Test
+    void unloadedUnitsAreTransferredToOnlyOneOfTheNewUnits() {
+      final Unit oldUnit = transport.create(1, player).get(0);
+      final List<Unit> newUnits = transport.create(2, player);
+
+      final List<Unit> unloadedInfantry = infantry.create(1, player);
+      oldUnit.setUnloaded(unloadedInfantry);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+      gameData.performChange(changes);
+
+      assertThat(newUnits.get(0).getUnloaded(), is(unloadedInfantry));
+      assertThat(
+          "Only the first new unit should get the unloaded infantry",
```

Comment:
Why is this the case? A more specific description could be the key, what is special about the new unit exactly that makes this assertion the case?